src/Fields/mapreduce.jl

```diff
@@ -5,6 +5,21 @@ weighted_jacobian(field) = weighted_jacobian(axes(field))
 weighted_jacobian(space::Spaces.AbstractSpace) =
     Spaces.local_geometry_data(space).WJ
 
+function _get_context(field)
+    space = axes(field)
+    topology =
+        space isa Spaces.ExtrudedFiniteDifferenceSpace ?
+        space.horizontal_space.topology : space.topology
+    topology isa Topologies.DistributedTopology2D ? topology.context : nothing
+end
+
+function _area(field)
+    context = _get_context(field)
+    area_local = Base.sum(weighted_jacobian(field))
+    isnothing(context) ? area_local :
+    ClimaComms.allreduce(context, area_local, +)
```

Comment:
see my other comment: if you change it so that `context` returns `SingletonCommsContext()` instead of `nothing`, then you don't need the branch.